node-backend-starter-code/client.js

```diff
@@ -0,0 +1,391 @@
+/* 
+
+I ♥ Movies Client
+==========
+Jeremiah Alexander
+
+*/
+
+
+var searchContainer, heartsContainer, searchInput, searchBtn, navHearts, navFavs, errorMessage, moviesContainer, uuid, shareLink;
+var omdbSearchUrl = "http://www.omdbapi.com/?s=";
+var omdbReadUrl = "http://www.omdbapi.com/?plot=full&i=";
+var includedDetails = ["Released", "Genre", "Director", "Writer", "Actors", "Metascore"];
+
+function setUUID(id){
+	
+	if ( !id || id === "" || id == uuid ) return;
+
+	uuid = id;
+
+	//update the browser url
+	var stateObj = { uuid: uuid };
+	history.pushState( stateObj, "I ♥ Movies || User Page", "?uuid=" + encodeURI(uuid) );
+	
+	//update the link to this page
+	shareLink.textContent = window.location;
+	shareLink.setAttribute("href", window.location);
+
+	//make hearts section visible
+	navHearts.className = "";
+}
+function displayError(message){
+	errorMessage.innerHTML = message || "";
+	if (!message){
+		errorMessage.className = "alert alert-danger hidden";
+	} else {
+		errorMessage.className = "alert alert-danger";
+	}
+}
+
+function clearMovies(){
+	//clear all listings - todo. could instead reuse elements rather than creating them again
+	while (moviesContainer.firstChild) 
+	{
+    	moviesContainer.removeChild(moviesContainer.firstChild);
+	}
+	searchInput.value = "";
+}
+
+//using the actual model id we remove a faviourite from our server
+function removeFromHearts( oid ){
+
+	var xmlhttpRequest = new XMLHttpRequest();
+
+	//send post request to save new faviourite
+	xmlhttpRequest.onreadystatechange = function(){
+		if ( xmlhttpRequest.readyState === 4 ){
+			//we'll ignore a success response and just out if we have an error
+			if (xmlhttpRequest.status !== 200){
+				displayError("Sorry but there was an error removing that heart, please reload the page and try again.");
+			}
+		}
+	};
+	xmlhttpRequest.open("DELETE", encodeURI("./favorites/" + oid), true);
+	xmlhttpRequest.send();
+}
+
+//we make an ajax request to our server to save a movie to faviourites
+function addToHearts( id, title ){
+
+	var xmlhttpRequest = new XMLHttpRequest();
```

Comment:
I've not worked with fetch but will investigate for the future. I had sort to rework it all with angular but just ran out of time. Whilst perhaps not respecting http verbs I thought a single toggle call might have worked too.